api-service/node/services/listBlockDevice.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2020 The OpenEBS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"context"
+	"strings"
+
+	"github.com/openebs/node-disk-manager/api-service/node"
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+	"github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	"github.com/openebs/node-disk-manager/pkg/hierarchy"
+	"github.com/openebs/node-disk-manager/pkg/util"
+	protos "github.com/openebs/node-disk-manager/spec/ndm"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/klog"
+)
+
+// Node helps in using types defined in package Node
+type Node struct {
+	node.Node
+}
+
+// NewNode returns an instance of type Node
+func NewNode() *Node {
+	return &Node{}
+}
+
+type disks []protos.BlockDevice
+
+//AllBlockDevices contains all the relationships and device types
+type AllBlockDevices struct {
+	Parents    []string
+	Partitions []string
+	LVMs       []string
+	RAIDs      []string
+	Holders    []string
+	Slaves     []string
+	Loops      []string
+	Sparse     []string
+}
+
+var all AllBlockDevices // This variable would contain all devices found in the node and their relationships
+
+// ListBlockDevices returns the block devices and their relationships
+func (n *Node) ListBlockDevices(ctx context.Context, null *protos.Null) (*protos.BlockDevices, error) {
+	klog.Info("Listing block devices")
+
+	ctrl, err := controller.NewController()
+	if err != nil {
+		klog.Errorf("Error creating a controller %v", err)
+		return nil, status.Errorf(codes.NotFound, "Namespace not found")
+	}
+
+	err = ctrl.SetControllerOptions(controller.NDMOptions{ConfigFilePath: "/host/node-disk-manager.config"})
```

Comment:
Make this path also into constant. 

Later we may need to make this configurable as done in the controller.